crates/matrix-sdk-ui/src/timeline/pagination.rs

```diff
@@ -52,20 +52,7 @@ impl super::Timeline {
                         trace!("Back-pagination succeeded with {num_events} events");
 
                         let handle_many_events_result =
-                            match self.inner.handle_back_paginated_events(events).await {
```

Comment:
I think that this will now have a bug in the logic? Since `events_received` still takes a `u16` but `num_events` is an `usize`. We will handle more than `u16` events but then we will abort before updating the `outcome` because `num_events` doesn't fit into `events_received`.

In other words we'll handle the events just fine, but return early and log that we aborted because of an overflow.